pygmt/datasets/map_tiles.py

```diff
@@ -0,0 +1,110 @@
+"""
+Function to load raster basemap tiles from XYZ tile providers, and load as
+:class:`xarray.DataArray`.
+"""
+
+try:
+    import contextily
+except ImportError:
+    contextily = None
+
+import numpy as np
+import xarray as xr
+
+__doctest_requires__ = {("load_map_tiles"): ["contextily"]}
+
+
+def load_map_tiles(region, source=None, lonlat=True, **kwargs):
+    """
+    Load a georeferenced raster basemap from XYZ tile providers.
+
+    The tiles that compose the map are merged and georeferenced into an
+    :class:`xarray.DataArray` image with 3 bands (RGB). Note that the returned
+    image is in a Spherical Mercator (EPSG:3857) coordinate reference system.
+
+    Parameters
+    ----------
+    region : list
+        The bounding box of the map in the form of a list [*xmin*, *xmax*,
+        *ymin*, *ymax*]. These coordinates should be in longitude/latitude if
+        ``lonlat=True`` or Spherical Mercator (EPSG:3857) if ``lonlat=False``.
+
+    source : xyzservices.TileProvider or str
+        [Optional. Default: Stamen Terrain web tiles] The tile source: web tile
+        provider or path to a local file. The web tile provider can be in the
+        form of a :class:`xyzservices.TileProvider` object or a URL. The
+        placeholders for the XYZ in the URL need to be {x}, {y}, {z},
+        respectively. For local file paths, the file is read with
+        :doc:`rasterio <rasterio:index>` and all bands are loaded into the
+        basemap. IMPORTANT: tiles are assumed to be in the Spherical Mercator
+        projection (EPSG:3857).
+
+    lonlat : bool
+        [Optional. Default: True]. If False, coordinates in ``region`` are
+        assumed to be Spherical Mercator as opposed to lon/lat.
+
+    kwargs : dict
+        Extra keyword arguments to pass to :func:`contextily.bounds2img`.
```

Comment:
It seems like there are only a few `contextily.bounds2img` parameters that are not wrapped (`wait` and `max_retries`). Why are you leaving that as a kwarg option instead of adding default values?

Reply:
Using `kwargs` means that if the API of `contextily.bounds2img` changes for any reason (e.g. new parameters added, or parameters get deleted or renamed), the `pygmt.datasets.load_map_tiles` function would still work and be cross-compatible with different `contextily` versions.

That said, the API does seem rather stable looking at https://github.com/geopandas/contextily/blame/v1.2.0/contextily/tile.py#L157-L195, so I could add them in to allow for tab-completion :slightly_smiling_face:

Reply:
Does the `kwargs` parameter still needed to be included? It looks like all available options have been wrapped.

Reply:
It seems the [`bounds2img` API documentation](https://contextily.readthedocs.io/en/latest/reference.html#contextily.bounds2img) doesn't show all the available parameters. For example, the tutorial (https://contextily.readthedocs.io/en/latest/intro_guide.html#Fine-tune-zoom-levels) also mentions the `zoom` parameter, which is not listed in the `bounds2img` API documentatin.

Reply:
Hmm yeah, `zoom` isn't in the API docs, but I see it in the code at https://github.com/geopandas/contextily/blob/6d479a6d2b616c3a1c216befd2736dd30eb355b5/contextily/tile.py#L174-L175. Should I add the `zoom` parameter then, or stick with `*kwargs`?

Reply:
> It seems the [`bounds2img` API documentation](https://contextily.readthedocs.io/en/latest/reference.html#contextily.bounds2img) doesn't show all the available parameters. For example, the tutorial ([contextily.readthedocs.io/en/latest/intro_guide.html#Fine-tune-zoom-levels](https://contextily.readthedocs.io/en/latest/intro_guide.html#Fine-tune-zoom-levels)) also mentions the `zoom` parameter, which is not listed in the `bounds2img` API documentatin.

Likely the [`bounds2img` API documentation](https://contextily.readthedocs.io/en/latest/reference.html#contextily.bounds2img) has changed since my last comment. The new documentation now shows the `zoom` parameter.

My concern with `**kwargs` is that users may pass `bounds2img`'s parameters `w`, `e`, `s`, `n` or `ll` to the `load_map_tiles` function.

Reply:
> My concern with `**kwargs` is that users may pass `bounds2img`'s parameters `w`, `e`, `s`, `n` or `ll` to the `load_map_tiles` function.

Fair point. I've added the `zoom` parameter and removed `**kwargs` in d06e2c0ee457c9bdbaa5d3de5b6366f10b754edb.